### PR TITLE
[Refactor][heartbeat] Make get fe heart response by thrift

### LINKED ISSUE #8037: [Enhancement] Make get fe heartbeat response more stable by use thrift as default method
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

as issue #5021 mentioned, get fe heartbeat response by http is unstable sometimes, so we provide rpc method to get fe heartbeat response, for compatibility, we don't remove http method, but now, we can make get fe heartbeat response more stable by use thrift as default method

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #8037 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (Yes)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Please create an issue to describe why do this and what is your motivation. Especially for this kind fo API change.

**Comment by caiconghui** [comment]
> Please create an issue to describe why do this and what is your motivation. Especially for this kind fo API change.

done

File: fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
  * {"status":"OK","msg":"Success","replayedJournal"=123456, "queryPort"=9000, "rpcPort"=9001}
  * {"status":"FAILED","msg":"err info..."}
  */
+@Deprecated
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: Why deprecate this?

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: > Why deprecate this?

get heartbeat info by rpc method is a more stable way, once get heartbeat by thrift, BootstrapFinishAction is not needed anymore, here we just keep compatible when fe upgrade

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: I thought that `BootstrapFinishAction` is nothing to do with the heartbeat?
It is used for checking whether the FE is started successfully.

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: > I thought that `BootstrapFinishAction` is nothing to do with the heartbeat? It is used for checking whether the FE is started successfully.

 controller.registerHandler(HttpMethod.GET, "/api/bootstrap", new BootstrapFinishAction(controller))
it is now used for heartbeat

   private HeartbeatResponse getHeartbeatResponseByHttp() {
            String url = "http://" + fe.getHost() + ":" + Config.http_port
                    + "/api/bootstrap?cluster_id=" + clusterId + "&token=" + token;

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: Oh, I forgot. But this API is also used for checking whether FE is started.
So we still need it. Just not use it for heartbeat.

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java:40]: And by the way, most of user does not switch the heartbeat method to thrift way. So maybe we need notice the user by somehow.

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/http/rest/BootstrapFinishAction.java]: Ok, I see

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**